server/src/main/java/org/apache/gravitino/server/web/filter/authorization/AssociateTagAuthorizationExecutor.java

```diff
@@ -0,0 +1,118 @@
+import java.util.Optional;
+import org.apache.gravitino.Entity;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.authorization.AuthorizationRequestContext;
+import org.apache.gravitino.dto.requests.TagsAssociateRequest;
+import org.apache.gravitino.server.authorization.expression.AuthorizationExpressionEvaluator;
+import org.apache.gravitino.server.web.rest.MetadataObjectTagOperations;
+
+/**
+ * Metadata object authorization for {@link
+ * MetadataObjectTagOperations#associateTagsForObject(String, String, String, TagsAssociateRequest)}
+ */
+public class AssociateTagAuthorizationExecutor implements AuthorizationExecutor {
+
+  private final Parameter[] parameters;
+  private final Object[] args;
+  private final Map<Entity.EntityType, NameIdentifier> metadataContext;
+  private final AuthorizationExpressionEvaluator authorizationExpressionEvaluator;
+  private final Map<String, Object> pathParams;
+  private final String entityType;
+
+  public AssociateTagAuthorizationExecutor(
+      Parameter[] parameters,
+      Object[] args,
+      Map<Entity.EntityType, NameIdentifier> metadataContext,
+      AuthorizationExpressionEvaluator authorizationExpressionEvaluator,
+      Map<String, Object> pathParams,
+      String entityType) {
+    this.parameters = parameters;
+    this.args = args;
+    this.metadataContext = metadataContext;
+    this.authorizationExpressionEvaluator = authorizationExpressionEvaluator;
+    this.pathParams = pathParams;
+    this.entityType = entityType;
+  }
+
+  @Override
+  public boolean execute() throws Exception {
+    Object request = extractFromParameters(parameters, args);
+    if (request == null) {
+      return false;
+    }
+
+    AuthorizationRequestContext context = new AuthorizationRequestContext();
+    Entity.EntityType targetType =
+        Entity.EntityType.TAG; // Tags are the only supported batch target here
+    Preconditions.checkArgument(request instanceof TagsAssociateRequest);
+    TagsAssociateRequest tagsAssociateRequest = (TagsAssociateRequest) request;
+    tagsAssociateRequest.validate();
+    // Authorize both 'tagsToAdd' and 'tagsToRemove' fields.
+    return authorizeTag(tagsAssociateRequest.getTagsToAdd(), context, targetType)
+        && authorizeTag(tagsAssociateRequest.getTagsToRemove(), context, targetType);
+  }
+
+  /**
+   * Performs batch authorization for a given field (e.g., "tagsToAdd" or "tagsToRemove") containing
+   * an array of tag names.
+   *
+   * @param tagNames tagNames
+   * @param context The shared authorization request context.
+   * @param targetType The entity type being authorized (expected to be TAG).
+   * @return {@code true} if all tags in the field pass authorization; {@code false} otherwise.
+   */
+  private boolean authorizeTag(
+      String[] tagNames, AuthorizationRequestContext context, Entity.EntityType targetType) {
+
+    // Treat null or empty arrays as no-op (implicitly authorized)
+    if (tagNames == null) {
```

Comment:
After the request's validation, we don't need check null value here. You can simplify the code

Reply:
They can't be null. If the tag names is null. We will fail to deseriliaze the data. If the value can null, we should mark it as @Nullable. The value can only be empty.